netbox/dcim/models/device_components.py

```diff
@@ -953,12 +953,13 @@ def clean(self):
         super().clean()
 
         # Check that positions count is greater than or equal to the number of associated FrontPorts
-        frontport_count = self.frontports.count()
-        if self.positions < frontport_count:
-            raise ValidationError({
-                "positions": f"The number of positions cannot be less than the number of mapped front ports "
-                             f"({frontport_count})"
-            })
+        if self.pk:
```

Comment:
Is this change related to [this Django bug](https://code.djangoproject.com/ticket/33952)? I don't have any problem with the change itself but I suspect there are similar instances elsewhere in the code that will need to be addressed.

Reply:
I don't think it is a Django "bug" it is more being more explicit about reporting an invalid condition.  This should probably only occur in the clean methods in our codebase - we are querying the m2m on the model when the pk isn't set yet, which doesn't really make any sense.  I think previously Django probably defaulted this to be None or 0 for count(), but now is explicitly raising that this shouldn't be done.